src/Microsoft.Unity.Analyzers/ImproperSerializeField.cs

```diff
@@ -0,0 +1,183 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Unity.Analyzers.Resources;
+
+namespace Microsoft.Unity.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class ImproperSerializeFieldAnalyzer : DiagnosticAnalyzer
+	{
+		public const string Id = "UNT0013";
+
+		private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			Id,
+			title: Strings.ImproperSerializeFieldDiagnosticTitle,
+			messageFormat: Strings.ImproperSerializeFieldDiagnosticMessageFormat,
+			category: DiagnosticCategory.Correctness,
+			defaultSeverity: DiagnosticSeverity.Info,
+			isEnabledByDefault: true,
+			description: Strings.ImproperSerializeFieldDiagnosticDescription);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.RegisterSyntaxNodeAction(AnalyzePropertyDeclaration, SyntaxKind.PropertyDeclaration);
+			context.RegisterSyntaxNodeAction(AnalyzeFieldDeclaration, SyntaxKind.FieldDeclaration);
+		}
+
+		private static void AnalyzePropertyDeclaration(SyntaxNodeAnalysisContext context)
+		{
+			var property = (MemberDeclarationSyntax)context.Node;
```

Comment:
Perhaps we can use instead ?
```csharp
			var property = (PropertyDeclarationSyntax)context.Node;
			if (property == null)
				return;
```

Reply:
You're right. This is left over from when I was trying to analyze both Properties and Fields in the same method. Turns out to be a large difference than I thought!